src/spox/_attributes.py

```diff
@@ -32,15 +32,28 @@
 
 class Attr(ABC, Generic[T]):
     _value: Union[T, "_Ref[T]"]
+    _name: str
     _attribute_proto_type_int: ClassVar[int]
+    _cached_onnx: Optional[AttributeProto]
 
-    def __init__(self, value: Union[T, "_Ref[T]"]):
+    def __init__(self, value: Union[T, "_Ref[T]"], name: str):
         self._value = value
+        self._name = name
+        self._cached_onnx = None
+
         self._validate()
 
+    def recreate_if_not_cachable(self, name: str) -> "Attr":
+        if isinstance(self, (AttrGraph, AttrType, _Ref)) or isinstance(
+            self._value, _Ref
+        ):
```

Comment:
This looks a bit suspicious to me now. Can we not have infinite-depth nesting? As in, should we also not check `self._value._value` and so on?

Reply:
It is rather convoluted... The property `self.value` (used below) will call `_deref` which will recurse. I think this is ok?